lib/Sema/TypeCheckStorage.cpp

```diff
@@ -341,8 +341,16 @@ IsSetterMutatingRequest::evaluate(Evaluator &evaluator,
   // or not based on the composition of the wrappers.
   if (auto var = dyn_cast<VarDecl>(storage)) {
     if (auto mut = var->getPropertyWrapperMutability()) {
-      return mut->Setter == PropertyWrapperMutability::Mutating
-        && result;
+      bool isMutating = mut->Setter == PropertyWrapperMutability::Mutating;
+      if (auto *accessor = var->getParsedAccessor(AccessorKind::DidSet)) {
+        // If there's a didSet, we call the getter for the 'oldValue', and so
+        // should consider the getter's mutatingness as well
+        if (!accessor->isSimpleDidSet()) {
```

Comment:
Is there a similar issue with `willSet`?

Reply:
I don't think so - the getter is called for the argument to `didSet`, while `willSet` just takes the setter argument. However, there's probably a similar issue wrt the mutability of `willSet`